zebra-chain/src/block/height.rs

```diff
@@ -65,60 +65,70 @@ impl Height {
 }
 
 /// A difference between two [`Height`]s, possibly negative.
-pub type HeightDiff = i32;
+///
+/// This can represent the difference between any height values,
+/// even if they are outside the valid height range (for example, in buggy RPC code).
+pub type HeightDiff = i64;
+
+impl TryFrom<u32> for Height {
+    type Error = &'static str;
+
+    /// Checks that the `height` is within the valid [`Height`] range.
+    fn try_from(height: u32) -> Result<Self, Self::Error> {
+        // Check the bounds.
+        if Height::MIN.0 <= height && height <= Height::MAX.0 {
```

Comment:
Nitpick/Optional: the first part of this condition will always be true for a u32

```suggestion
        if height <= Height::MAX.0 {
```

Reply:
Yep, but we want it in there in case we change the underlying type.